src/report/report-generate.controller.js

```diff
@@ -113,5 +130,44 @@
                 '_blank'
             );
         }
+
+
+        /**
+         * @ngdoc method
+         * @methodOf report.controller:ReportGenerateController
+         * @name watchDependency
+         *
+         * @description
+         * Sets up a watch on report parameter selection,
+         * to update dependent parameters options based on it's value.
+         */
+        function watchDependency(param, dep) {
```

Comment:
Well, when looping through each parameter, we don't know whether the evaluated parameter is a dependency for any other parameter. What we do know is what are dependencies for a given parameter, and I believe assigning those in html code would be at best very messy.